Better CI setup for simple understanding of failed tests

whyoleg opened the pull request:
### Motivation:

On current moment, it's hard to understand which test, for which target and why it failed.

### Modifications:

* split running tests per target
* integrate `mikepenz/action-junit-report` github action to show report about failed tests and stacktraces
* run jvm tests on 8, 11 and 15 JVM versions
* split tests in 3 groups: core, local, ktor
* added timeouts for tests - sometimes ktor K/N tests are just freeze - still WIP why
* turn on debug logs of test runner on CI - to understand tests freezing
* better exception if test fails in before/after

### Result:

Easier to understand, what to fix.
Still, CI tests are unstable (locally jvm/js/mac tests works ok)

yschimke commented on .github/workflows/benchmarks.yml:
```diff
@@ -6,19 +6,34 @@ jobs:
-          java-version: 11
+          distribution: 'adopt'
+          java-version: 15
+      - uses: actions/cache@v2
```
Curious whether instead of caching the CI run, we could run an RSocket https://hub.docker.com/r/gradle/build-cache-node/ and use for speeding up local development.  Will sync with you in slack.

yschimke commented on rsocket-test/src/commonMain/kotlin/io/rsocket/kotlin/test/TransportTest.kt:
```diff
@@ -152,7 +152,7 @@ abstract class TransportTest : SuspendTest {
 
     @Test
-    fun requestResponse100000() = test(TransportTestLongDuration) {
+    fun requestResponse100000() = test(10.minutes) {
```
I suspect we want these comfortably inside 1 minute for reliable CI runs.

whyoleg replied:
K/N is just not so fast enough :(

yschimke reviewed (approved):
low signal review

whyoleg commented:
**UPDATE:**
* removed `Tomcat` websocket test - it's slow, ktor should have sufficient tests to make sure, that engines WS implementation is the same. CIO/Netty/Jetty left, as they are anyway fast and popular.
* ignored 3 long tests on K/N to speed up tests - now all tests are executed in less than 20 minutes
* removed `gradle-all` workflow - when doing PR it creates runs double of jobs needed. If needed to run tests for specific branch - `Run tests` workflow can be used from Github UI.
